Upgrade golang.org/x/net and golang.org/x/sys

paskozdilar opened the pull request:
Fixes https://github.com/irai/packet/issues/15.

irai commented:
Thanks for investigating this. As you pointed out, the issue seems to be related to go 1.23 and deep in the golang.org/x/sys package.

My initial thought is to upgrade all dependencies to support the latest /x/sys package as this is likely fixed in the package itself.

There are many security and other issues in earlier packages so I will want to avoid a downgrade.

Do you want to try upgrading all dependencies instead? I would welcome a patch for that.

paskozdilar committed:
Update dependencies with "go get -u && go mod tidy"

paskozdilar commented:
Indeed, seems like upgraded dependencies work too.

For some reason, `./handlers/dns_naming` tests seem to hang and spam the terminal:
```
mdns  : "NSEC resource not implemented" name="Goth.local." hdr={Name:Goth.local. Type:47 Class:32769 TTL:120 Length:8}
mdns  : "NSEC resource not implemented" name="Goth.local." hdr={Name:Goth.local. Type:47 Class:32769 TTL:120 Length:8}
mdns  : "NSEC resource not implemented" name="Goth.local." hdr={Name:Goth.local. Type:47 Class:32769 TTL:120 Length:8}
[...]

panic: test timed out after 10s
        running tests:
                TestMDNSHandler_Apple (6s)
                TestMDNSHandler_Apple/MacBook (6s)

goroutine 68 [running]:
testing.(*M).startAlarm.func1()
        /usr/lib/go-1.23/src/testing/testing.go:2373 +0x385
created by time.goFunc
        /usr/lib/go-1.23/src/time/sleep.go:215 +0x2d

goroutine 1 [chan receive]:
testing.(*T).Run(0xc00012a680, {0x68a9a8?, 0x0?}, 0x6b1b30)
        /usr/lib/go-1.23/src/testing/testing.go:1751 +0x3ab
testing.runTests.func1(0xc00012a680)
        /usr/lib/go-1.23/src/testing/testing.go:2168 +0x37
testing.tRunner(0xc00012a680, 0xc00008dc70)
        /usr/lib/go-1.23/src/testing/testing.go:1690 +0xf4
testing.runTests(0xc000124270, {0x8bd280, 0xb, 0xb}, {0x46faf0?, 0x46f75a?, 0x8c41e0?})
        /usr/lib/go-1.23/src/testing/testing.go:2166 +0x43d
testing.(*M).Run(0xc000118b40)
        /usr/lib/go-1.23/src/testing/testing.go:2034 +0x64a
main.main()
        _testmain.go:67 +0x9b

goroutine 29 [chan receive]:
testing.(*T).Run(0xc000190340, {0x68575d?, 0xc000106c40?}, 0xc00007a1a0)
        /usr/lib/go-1.23/src/testing/testing.go:1751 +0x3ab
github.com/irai/packet/handlers/dns_naming.TestMDNSHandler_Apple(0xc000190340)
        ${REDACTED}/go/packet/handlers/dns_naming/mdns_test.go:333 +0x246
testing.tRunner(0xc000190340, 0x6b1b30)
        /usr/lib/go-1.23/src/testing/testing.go:1690 +0xf4
created by testing.(*T).Run in goroutine 1
        /usr/lib/go-1.23/src/testing/testing.go:1743 +0x390

goroutine 31 [select]:
github.com/irai/packet.Config.NewSession.func2(0xc000038320)
        ${REDACTED}/go/packet/session.go:199 +0x9a
created by github.com/irai/packet.Config.NewSession in goroutine 29
        ${REDACTED}/go/packet/session.go:196 +0x7c8

goroutine 30 [select]:
github.com/irai/packet.Config.NewSession.func1(0xc000038320)
        ${REDACTED}/go/packet/session.go:178 +0x8e
created by github.com/irai/packet.Config.NewSession in goroutine 29
        ${REDACTED}/go/packet/session.go:175 +0x771

goroutine 32 [chan receive]:
github.com/irai/packet.(*bufferedPacketConn).ReadFrom(0x0?, {0xc0000bc800, 0x0?, 0x7d0})
        ${REDACTED}/go/packet/memconn.go:36 +0x39
github.com/irai/packet.TestReadAndDiscardLoop({0x6f7610, 0xc00002c410})
        ${REDACTED}/go/packet/memconn.go:56 +0x5e
created by github.com/irai/packet/handlers/dns_naming.TestMDNSHandler_Apple in goroutine 29
        ${REDACTED}/go/packet/handlers/dns_naming/mdns_test.go:307 +0xbf

goroutine 33 [runnable]:
syscall.Syscall(0x1, 0x2, 0xc00013fb00, 0x77)
        /usr/lib/go-1.23/src/syscall/syscall_linux.go:73 +0x25
syscall.write(0xc0001340c0?, {0xc00013fb00?, 0xd?, 0x1ad?})
        /usr/lib/go-1.23/src/syscall/zsyscall_linux_amd64.go:964 +0x3b
syscall.Write(...)
        /usr/lib/go-1.23/src/syscall/syscall_unix.go:211
internal/poll.ignoringEINTRIO(...)
        /usr/lib/go-1.23/src/internal/poll/fd_unix.go:745
internal/poll.(*FD).Write(0xc0001340c0, {0xc00013fb00, 0x77, 0x800})
        /usr/lib/go-1.23/src/internal/poll/fd_unix.go:381 +0x394
os.(*File).write(...)
        /usr/lib/go-1.23/src/os/file_posix.go:46
os.(*File).Write(0xc0001220a8, {0xc00013fb00?, 0x77, 0xc00013fb3f?})
        /usr/lib/go-1.23/src/os/file.go:195 +0x51
github.com/irai/packet/fastlog.(*Line).Write(0xc00013fb00)
        ${REDACTED}/go/packet/fastlog/logging.go:226 +0x91
github.com/irai/packet/handlers/dns_naming.(*DNSHandler).ProcessMDNS(0xc0001b22d0, {{0x84e3e0, 0x215, 0x215}, 0xe, 0x0, 0x22
, 0x0, 0x2a, 0xd, ...})
        ${REDACTED}/go/packet/handlers/dns_naming/mdns.go:517 +0x20ed
github.com/irai/packet/handlers/dns_naming.TestMDNSHandler_Apple.func1(0xc0001904e0)
        ${REDACTED}/go/packet/handlers/dns_naming/mdns_test.go:344 +0x30a
testing.tRunner(0xc0001904e0, 0xc00007a1a0)
        /usr/lib/go-1.23/src/testing/testing.go:1690 +0xf4
created by testing.(*T).Run in goroutine 29
        /usr/lib/go-1.23/src/testing/testing.go:1743 +0x390
FAIL    github.com/irai/packet/handlers/dns_naming      10.017s
```

I will try to investigate this.

You can merge this PR if you'd like, since it's just go.mod & go.sum update, or wait until the cause of test failure is figured out. Your call.

paskozdilar commented:
The failing test seem to be `TestMDNSHandler_Apple` in `handlers/dns_naming/mdns_test.go`.

The test code didn't change since v0.4.0 (which passes), so I assume some other change in the code has caused this.
It will take me some time to get deeper into the codebase to figure this out.

irai reviewed (approved):
Great. 
This is a better solution than downgrading and will keep dependencies up to date.
